### ISSUE TITLE
PNext, ScErrDatabaseLangUnsupported

### ISSUE THREAD

**Issue by careri** [Starcounter/Starcounter#3821]
PnextNightlyBuilds-2.3.3694

The TurboText dlls in my installation are called 4, e.g:
>TurboText_en-GB_4.dll

From VS:
>
Error	1	Creating database Default failed due to an unexpected problem. ("Error invoking ""C:\Program Files\Starcounter\sccreatedb.exe" -ip "C:\Users\carle\OneDrive\Documents\Starcounter\Personal\Data\Default\Default-20160902T120302179" -coll TurboText_en-GB_3.dll -oid 1 -lastoid 4611686018427387903 Default": exit code is 0x1. Tool output:

>
 Starcounter permanent storage creation utility.
Copyright (C) Starcounter AB 2007-2016. All rights reserved.
>
Error: ScErrDatabaseLangUnsupported (SCERR2023): One of more of the languages the database is configured for isn't supported by the current system.
>
"). (ScErrUnexpectedCommandException (SCERR10016))
>
(Double-click here for additional help)		0	0

**Comment by careri** [comment]
I copied the assemblies and changed to version 3. Seems to work

**Comment by eriohl** [comment]
With 2.3 TurboText version 4 should and must be used. It won't work fine with TurboText 3. Is that command line generated by Starcounter? Then something is not working as it should.

**Comment by eriohl** [comment]
Could it be this incomaptible base config from previous installation causing problems again? cc @per-samuelsson

**Comment by careri** [comment]
The command are generated by starcounter. I only renamed my TurboText4 dlls to TurboText3, then it worked. I'll scan my configs

**Comment by careri** [comment]
You're probably right. The default config file:
Documents\Starcounter\Personal\Personal.server.config

Contains a reference to TurboText3. Might be that it's a file kept from previous installation of sc2.2

**Comment by dan31** [comment]
@careri while testing nightly builds, this is indeed sometimes unsafe to keep the old config files around. At release time, we will instruct how to address changes in these old files, migrate data (2.2 and 2.3 are not compatible at disk storage format) and address other changes. Until the release time - sorry for inconvenience!

@eriohl is that fine if we just remove TT3 files from 2.3 distribution? CC @alemoi @bigwad

**Comment by eriohl** [comment]
I think TT3 is already removed from the installation (which is why operation failed initially). If not they should be. 2.3 does not work with TT3.

Yes @careri: It will work if you rename the DLL because then you are actually using TT4. I got the impression that you copied the DLLs from a previous installation or something.

**Comment by careri** [comment]
Yepp, that was why I tried renaming. Sorry for being unclear.

**Comment by bigwad** [comment]
It has happened once again. This time to @RickardR. @alemoi, I think there is some bug with upgrade. Please take a look.

**Comment by RickardR** [comment]
Yeah, seems Personal.server.config wasn't updated in the upgrade.
Hade to change 
<CollationFile>TurboText_en-GB_3.dll</CollationFile> manually to _4

**Comment by alemoi** [comment]
I think there is no upgrade of configs to TT4 when installing 2.3 on existing 2.2 installation. I will check the files also.

**Comment by per-samuelsson** [comment]
I don't know all the details here, but just a general advice: let's try to have things that are truly meaningful to customize configurable, and those that aren't more static.

How many users are likely (at *this stage in our development*) to customize what Turbo Text library to use? How many such libraries (with different languages) do we even support as of today? Very few I suppose.

Can we release the maintenance burden simply by removing this thing from configuration and have it hard coded instead? Alternatively, we presume there is no configuration and use the one that best fit the current version if no configuration is given?

**Comment by alemoi** [comment]
I agree, the language is needed to be configurable, but not the version, I suppose. What would be the need to switch between versions? (unless you upgrade the version, but then its unload/reload I suppose)

**Comment by per-samuelsson** [comment]
> What would be the need to switch between versions?

Not sure I understand that question? And not sure I would be able to answer it even if I did. 😄 

My basic thinking was that we supported localization using a moniker and give ourselves the freedom to map that to whatever internal bits we find suitable for a given version. Example: if nothing is given, we suppose moniker "en_GB". On top of that, we allow any user to *add* a configuration, overwriting the default (we don't even show it's possible to configure it by default, since the absolute majority will never care anyway). Then for each version, we decide what the given moniker alias.

**Comment by alemoi** [comment]
> Not sure I understand that question? And not sure I would be able to answer it even if I did. 😄

I mean, if you have Starcounter installed, and have two versions of Turbotext, e.g. 3 and 4. Why would you need to have 2 versions? I see the problem can arise when version was upgraded silently with a new installation but no unload/reload was made.

**Comment by k-rus** [comment]
I am wonder what happens if a database was filled with data using TT3 and then new Starcounter was installed with TT4? Will the database start up fail? Will it be meaningful error?
